compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -2588,7 +2588,7 @@ impl<'a> Parser<'a> {
     }
 
     /// Parses the condition of a `if` or `while` expression.
-    fn parse_expr_cond(&mut self) -> PResult<'a, P<Expr>> {
+    pub fn parse_expr_cond(&mut self) -> PResult<'a, P<Expr>> {
```

Comment:
Please add a comment with links to projects where it will be used, otherwise this will get reverted again